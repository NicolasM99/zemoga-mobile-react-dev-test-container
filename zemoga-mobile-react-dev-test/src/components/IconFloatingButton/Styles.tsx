import { StyleSheet } from 'react-native';

import { COLORS } from 'src/constants/theme/colors';

export const iconFloatingButtonStyles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    bottom: 16,
    right: 16,
    borderRadius: 50,
    backgroundColor: COLORS.primary,
    height: 64,
    width: 64,
    justifyContent: 'center',
    alignItems: 'center'
  },
  disabled: {
    opacity: 0.5
  }
});
