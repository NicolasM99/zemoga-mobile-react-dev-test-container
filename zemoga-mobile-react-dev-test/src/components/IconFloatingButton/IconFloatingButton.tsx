import React, { FC } from 'react';

import { FontAwesome5 } from '@expo/vector-icons';
import { View, TouchableOpacity } from 'react-native';

import { IIconFloatingButton } from './@types/iconFloatingButton';
import { iconFloatingButtonStyles } from './Styles';

const IconFloatingButton: FC<IIconFloatingButton> = ({ icon, onPress }: IIconFloatingButton) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={iconFloatingButtonStyles.container}>
        <FontAwesome5 name={icon} size={32} color={'white'} />
      </View>
    </TouchableOpacity>
  );
};

export default IconFloatingButton;
