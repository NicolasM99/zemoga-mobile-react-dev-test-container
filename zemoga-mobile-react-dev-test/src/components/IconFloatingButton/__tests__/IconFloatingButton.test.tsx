import { render, screen } from '@testing-library/react-native';

import IconFloatingButton from '../IconFloatingButton';

describe('<IconFloatingButton /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<IconFloatingButton icon="comment" />);
    expect(screen).toMatchSnapshot();
  });
});
