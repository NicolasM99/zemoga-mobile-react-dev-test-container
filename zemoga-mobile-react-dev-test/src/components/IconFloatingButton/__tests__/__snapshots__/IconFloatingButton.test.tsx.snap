// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconFloatingButton /> unit tests Should mount correctly the component 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#10c64e",
        "borderRadius": 50,
        "bottom": 16,
        "flex": 1,
        "height": 64,
        "justifyContent": "center",
        "opacity": 0.5,
        "position": "absolute",
        "right": 16,
        "width": 64,
      }
    }
  >
    <Text />
  </View>
</View>
`;
